### PR TITLE
Repair deprecated fetch repodata

### LINKED ISSUE #13595: fetch_repodata_remote_request raises RepodataOnDisk exception
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

The `fetch_repodata_remote_request` was deprecated in 23.9, and is scheduled for removal in conda 24.3. Unfortunately, it it seems that it is now broken:

```
>>> from conda.core.subdir_data import fetch_repodata_remote_request
>>> fetch_repodata_remote_request('https://repo.anaconda.com/pkgs/main/linux-64', None, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/deprecations.py", line 71, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/core/subdir_data.py", line 560, in fetch_repodata_remote_request
    raw_repodata_str = subdir._repo.repodata(cache_state)  # type: ignore
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/jlap/interface.py", line 54, in repodata
    self.repodata_parsed(state)
  File "/Users/mgrant/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/jlap/interface.py", line 121, in repodata_parsed
    raise RepodataOnDisk()
conda.gateways.repodata.RepodataOnDisk
```

### Conda Info

```shell
active environment : base
    active env location : /Users/mgrant/miniconda3
            shell level : 1
       user config file : /Users/mgrant/.condarc
 populated config files : /Users/mgrant/.condarc
          conda version : 24.1.1
    conda-build version : 3.28.3
         python version : 3.11.7.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=m1
                          __conda=24.1.1=0
                          __osx=14.3=0
                          __unix=0=0
       base environment : /Users/mgrant/miniconda3  (writable)
      conda av data dir : /Users/mgrant/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/mgrant/miniconda3/pkgs
                          /Users/mgrant/.conda/pkgs
       envs directories : /Users/mgrant/miniconda3/envs
                          /Users/mgrant/.conda/envs
               platform : osx-arm64
             user-agent : conda/24.1.1 requests/2.31.0 CPython/3.11.7 Darwin/23.3.0 OSX/14.3 solver/libmamba conda-libmamba-solver/23.12.0 libmambapy/1.5.6 aau/0.4.3
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /Users/mgrant/.condarc <==
anaconda_anon_usage: True
```

### Conda list

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dholth**
### Description

Fix #13595 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Review by kenodegard — CHANGES_REQUESTED** [review]
please rebase and target the `24.1.x` branch instead

**Comment by dholth** [comment]
We could add a test similar to the jlap tests that fetch from a local web server, instead of the only remaining test for this function that fetches from a non-existing file. On the other hand we plan to remove this next release.

**Comment by kenodegard** [comment]
@dholth if it's easy to add (<30min?) I say we add it as a sanity check even if it gets removed again in March

**Comment by dholth** [comment]
@kenodegard I've added a test, manually checking that it fails with the old code but passes with `repo_interface_cls=CondaRepoInterface,`

File: tests/core/test_subdir_data.py [hunk 1]
```diff
@@ -154,6 +156,16 @@ def test_fetch_repodata_remote_request_invalid_arch():
     assert result is None
 
 
+def test_fetch_repodata_remote_request(package_server: socket, tmp_path: Path, mocker):
```
Comment (kenodegard) [inline tests/core/test_subdir_data.py:159]:
```suggestion
def test_fetch_repodata_remote_request(package_server: socket):
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

**Review by kenodegard — APPROVED** [review]

**Pull request merged**